bindings/pyroot_experimental/pythonizations/test/CMakeLists.txt

```diff
@@ -4,116 +4,126 @@
-
-# TIter pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_titer_iterator titer_iterator.py)
-
-# TClonesArray pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tclonesarray_itemaccess tclonesarray_itemaccess.py)
-
-# TArray and subclasses pythonizations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tarray_len tarray_len.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tarray_getitem tarray_getitem.py)
-
-# TVectorT pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tvectort_len tvectort_len.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tvectort_getitem tvectort_getitem.py)
-
-# TVector3 pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tvector3_len tvector3_len.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tvector3_getitem tvector3_getitem.py)
-
-# TString pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tstring_len tstring_len.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tstring_str_repr tstring_str_repr.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tstring_comparisonops tstring_comparisonops.py)
-
-# TString converter
-ROOT_ADD_PYUNITTEST(pyroot_conv_tstring tstring_converter.py)
-
-# TObjString pythonisations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tobjstring_len tobjstring_len.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tobjstring_str_repr tobjstring_str_repr.py)
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tobjstring_comparisonops tobjstring_comparisonops.py)
-
-# RVec and subclasses pythonizations
-ROOT_ADD_PYUNITTEST(pyroot_pyz_rvec_asrvec rvec_asrvec.py DEPENDENCIES_FOUND ${NUMPY_FOUND})
-
-# RDataFrame and subclasses pythonizations
-if (dataframe)
-    ROOT_ADD_PYUNITTEST(pyroot_pyz_rdataframe_asnumpy rdataframe_asnumpy.py DEPENDENCIES_FOUND ${NUMPY_FOUND})
-    ROOT_ADD_PYUNITTEST(pyroot_pyz_rdataframe_makenumpy rdataframe_makenumpy.py DEPENDENCIES_FOUND ${NUMPY_FOUND})
-endif()
-
-# RTensor pythonizations
-if (dataframe)
-    ROOT_ADD_PYUNITTEST(pyroot_pyz_rtensor rtensor.py DEPENDENCIES_FOUND ${NUMPY_FOUND})
-endif()
-
-# Passing Python callables to ROOT.TF
-ROOT_ADD_PYUNITTEST(pyroot_pyz_tf_pycallables tf_pycallables.py)
-
-if(roofit)
-  # RooAbsCollection and subclasses pythonizations
-  ROOT_ADD_PYUNITTEST(pyroot_pyz_rooabscollection_len rooabscollection_len.py)
-
-  # RooDataHist pythonisations
-  ROOT_ADD_PYUNITTEST(pyroot_pyz_roodatahist_ploton roodatahist_ploton.py)
-endif()
-
-# std::string_view backport in CPyCppyy
-if (dataframe)
-    ROOT_ADD_PYUNITTEST(pyroot_string_view_backport string_view_backport.py)
-endif()
-
-# Test wrapping Python callables for use in C++
-if (dataframe)
-    ROOT_ADD_PYUNITTEST(pyroot_cppcallable cppcallable.py)
-endif()
-
-# TComplex pythonizations
-ROOT_ADD_PYUNITTEST(pyroot_tcomplex tcomplex_operators.py)
+list(LENGTH python_executables len)
+math(EXPR how_many_pythons "${len} - 1")
+
+foreach(val RANGE ${how_many_pythons})
+  list(GET python_executables ${val} python_executable)
+  list(GET python_version_majors ${val} py_maj)
+  list(GET numpys_found ${val} numpy_found)
+
+  # Test ROOT module
+  ROOT_ADD_PYUNITTEST(pyroot_root_module_python${py_maj} root_module.py PY_EX ${python_executable})
```

Comment:
Since the other argument names are already quite long, to keep the same style I would call this argument `PYTHON_EXECUTABLE` instead of `PY_EX`.

That means that maybe it would be good to split the definition of the test in multiple lines (the PYTHON_EXECUTABLE argument can be in another line, and for the ones that have the DEPENDENCIES_FOUND, another line for that arg too).